mlrun/artifacts/document.py

```diff
@@ -216,14 +220,15 @@ def __init__(
 
     def __init__(
         self,
-        key=None,
-        document_loader: DocumentLoaderSpec = DocumentLoaderSpec(),
+        original_source: Optional[str] = None,
+        document_loader: Optional[DocumentLoaderSpec] = None,
         **kwargs,
     ):
-        super().__init__(key, **kwargs)
-        self.spec.document_loader = document_loader.to_str()
-        if "src_path" in kwargs:
-            self.spec.original_source = kwargs["src_path"]
+        super().__init__(**kwargs)
+        self.spec.document_loader = (
+            document_loader.to_str() if document_loader else self.spec.document_loader
```

Comment:
Why `to_str()`? You're using `from_dict` to de-serialize, but in general - why not maintain the object as object? You only need to serialize it when serializing the entire object.

Reply:
will fix